File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;

namespace HowToChangeCellColor {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();

            DataTable table = new DataTable();
            table.Columns.Add("column", typeof(string));
            table.Columns.Add("data", typeof(int));
            table.Columns.Add("hidden", typeof(int));
            table.Rows.Add("col1", 1, 2);
            table.Rows.Add("col2", 1, -2);

            pivotGridControl1.DataSource = table;
        }

        private void pivotGridControl1_CustomDrawCell(object sender, DevExpress.XtraPivotGrid.PivotCustomDrawCellEventArgs e) {
            if(e.DataField == fieldData) {
                PivotDrillDownDataSource ds = e.CreateDrillDownDataSource();
                int hiddenValue = (int)ds.GetValue(0, "hidden");
                if(hiddenValue > 0)
                    e.Appearance.BackColor = Color.Green;
                if(hiddenValue < 0)
                    e.Appearance.BackColor = Color.Red;
            }
        }
    }
}

File: VB/Form1.vb
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid

Namespace HowToChangeCellColor

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            Dim table As DataTable = New DataTable()
            table.Columns.Add("column", GetType(String))
            table.Columns.Add("data", GetType(Integer))
            table.Columns.Add("hidden", GetType(Integer))
            table.Rows.Add("col1", 1, 2)
            table.Rows.Add("col2", 1, -2)
            pivotGridControl1.DataSource = table
        End Sub

        Private Sub pivotGridControl1_CustomDrawCell(ByVal sender As Object, ByVal e As PivotCustomDrawCellEventArgs)
            If e.DataField Is fieldData Then
                Dim ds As PivotDrillDownDataSource = e.CreateDrillDownDataSource()
                Dim hiddenValue As Integer = CInt(ds.GetValue(0, "hidden"))
                If hiddenValue > 0 Then e.Appearance.BackColor = Color.Green
                If hiddenValue < 0 Then e.Appearance.BackColor = Color.Red
            End If
        End Sub
    End Class
End Namespace
